src/main/java/net/atos/zac/flowable/TakenService.java

```diff
@@ -172,18 +172,19 @@ public TaskInfo readTask(final String taskId) {
 
     public Task readOpenTask(final String taskId) {
         final Task task = findOpenTask(taskId);
-        if (task == null) {
-            throw new RuntimeException(String.format("No open task found with task id '%s'", taskId));
-        }
-        return task;
+        return validateTaskExistence(taskId, task);
     }
 
     public HistoricTaskInstance readClosedTask(final String taskId) {
         final HistoricTaskInstance historicTaskInstance = findClosedTask(taskId);
-        if (historicTaskInstance == null) {
-            throw new RuntimeException(String.format("No closed task found with task id '%s'", taskId));
+        return validateTaskExistence(taskId, historicTaskInstance);
+    }
+
+    private <T> T validateTaskExistence(String taskId, T task) throws RuntimeException {
```

Comment:
To be honest I think a separate method just for this null checking is overkill. It does not improve the readability. I would not create this separate method at all. The name is also a little misleading because it not only validates the existence, it also returns it, which I find a little strange.

I would just leave the code as is to be honest? (and start converting it to Kotlin.. ;-) )

Reply:
How does this PR fix the issue at hand? Is that just the different exception message?

Reply:
Yep. Requesting https://zaakafhandelcomponent-zac-dev.dimpact.lifely.nl/taken/123456789 for example, reports that this taken is closed.

If it is closed, then it existed. I spent 15 minutes looking for "closed" task, when in fact this was a wrong id

Reply:
Removed the helper method